finality-aleph/src/party/backup.rs

```diff
@@ -118,27 +118,22 @@ pub fn rotate(
     Ok((backup_saver, backup_loader))
 }
 
-/// Removes the backup directory for a session.
+/// Removes the backup directory for all sessions except the current session.
 ///
 /// `backup_path` is the path to the backup directory (i.e. the argument to `--backup-saving-path`).
 /// If it is `None`, nothing is done.
 ///
-/// Any filesystem errors are logged and dropped.
+/// Any filesystem errors are returned.
 ///
-/// This should be done after the end of the session.
-pub fn remove(path: Option<PathBuf>, session_id: u32) {
-    let path = match path {
-        Some(path) => path.join(session_id.to_string()),
-        None => return,
-    };
-    match fs::remove_dir_all(path) {
-        Ok(()) => {
-            debug!(target: "aleph-party", "Removed backup for session {}", session_id);
-        }
-        Err(err) => {
-            if err.kind() != io::ErrorKind::NotFound {
-                warn!(target: "aleph-party", "Error cleaning up backup for session {}: {}", session_id, err);
+/// This should be done at the beginning of the new session.
+pub fn remove_old_backups(path: Option<PathBuf>, current_session: u32) -> io::Result<()> {
```

Comment:
Nit: How about `remove_other_backups` or such, as we don't explicitly assume there won't be any backups of newer sessions?

Reply:
It would actually be better if this only removed old sessions – we might not want to accidentally delete a backup if we had to e.g. start from a snapshot because of DB corruption or something.

Reply:
done